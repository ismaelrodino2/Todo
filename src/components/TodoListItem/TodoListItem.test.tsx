import { TodoListItem } from '.';

import { fireEvent, render, waitFor } from '@testing-library/react';

describe('Tests for TodoListItem component', () => {
  it('Should render a todo', async () => {
    const todoTest = { complete: false, id: 1644511379001, text: 'test' };

    const { getByText } = render(<TodoListItem todo={todoTest} />);

    expect(getByText).toMatchSnapshot();
  });
  it('Should delete todo when click the button', async () => {
    const todoTest = { complete: false, id: 1644511379001, text: 'test' };

    const { getByTestId: deleteButton, getByText } = render(
      <TodoListItem todo={todoTest} />,
    );

    const btnNode = await waitFor(() => deleteButton('delete-btn'));

    const listNode = await waitFor(() => getByText('test'));

    fireEvent.click(btnNode);

    expect(listNode).toBeDefined();
  });
});
describe('Tests for TodoList component', () => {
  it('Should match snapshot', async () => {
    const todoTest = { complete: false, id: 1644511379001, text: 'test' };

    const { container } = render(<TodoListItem todo={todoTest} />);

    expect(container).toMatchSnapshot();
  });
});
