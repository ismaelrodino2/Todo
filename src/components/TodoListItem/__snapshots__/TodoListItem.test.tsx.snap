// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for TodoList component Should match snapshot 1`] = `
<div>
  <div
    class="flex justify-center"
  >
    <div>
      <div
        class="form-check items-center flex"
      >
        <input
          class="form-check-input appearance-none h-4 w-4 border border-gray-300 rounded-sm bg-white checked:bg-blue-600 checked:border-blue-600 focus:outline-none transition duration-200 mt-1 align-top bg-no-repeat bg-center bg-contain float-left mr-2 cursor-pointer"
          data-testid="todo-value"
          type="checkbox"
          value="test"
        />
        <label
          class="form-check-label inline-block text-gray-800"
          style="text-decoration: none;"
        >
          test
        </label>
        <button
          class="bg-pink-400 ml-1 hover:bg-pink-500 text-white font-bold py-1 px-3 rounded-full"
          data-testid="delete-btn"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests for TodoListItem component Should render a todo 1`] = `[Function]`;
